docs-sphinx/user-guide/how-to-use-header-only-layoutbuilder.md

```diff
@@ -0,0 +1,260 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: C++14
+  language: C++14
+  name: xcpp14
+---
+
+How to use the header-only LayoutBuilder in C++
+===============================================
+
+What is header-only Layout Builder?
+-----------------------------------
+
+The header-only Layout Builder consists of a set of compile-time, templated, static C++ classes, implemented entirely in header file which can be dropped into any external project, and easily separable from the rest of the Awkward C++ codebase.
+
+The Layout Builder namespace consists of the following 14 types of Layout Builders:
+
+* `Numpy`
+* `ListOffset`
+* `List`
+* `Empty`
+* `EmptyRecord`
+* `Record`
+* `Tuple`
+* `Regular`
+* `Indexed`
+* `IndexedOption`
+* `Unmasked`
+* `ByteMasked`
+* `BitMasked`
+* `Union`
```

Comment:
This looks more like a reference guide. I think, would be better to add a link to the reference guide instead of listing them.